### PR TITLE
feat: templates view video carousel

### PR DESCRIPTION
**Pull request by Kneesal**
# Description

### <samp>🤖 Generated by Copilot at ad6eb5f</samp>

Added a video preview feature for journey templates. Created a new `TemplateVideoPreview` component that shows the video blocks in a swiper carousel using `video.js`. Added a new tab in `TemplatePreviewTabs` to render the `TemplateVideoPreview` component.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/34356579/todos/6701656090)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] it should display videos as a carousel 
- [ ] it should only display videos that are used in the card
- [ ] it should display duplicates e.g. if the same video is used in two different cards, should render two videos

# Walkthrough

### <samp>🤖 Generated by Copilot at ad6eb5f</samp>

*  Import `useMemo` hook and `GetJourney_journey_blocks_VideoBlock` type ([link](https://github.com/JesusFilm/core/pull/2002/files?diff=unified&w=0#diff-501634969741d25ebfcafc40c5ccc588acb4eaaeb397d9b5d56f541f7a38c08fL8-R8), [link](https://github.com/JesusFilm/core/pull/2002/files?diff=unified&w=0#diff-501634969741d25ebfcafc40c5ccc588acb4eaaeb397d9b5d56f541f7a38c08fL17-R23))
*  Create `TemplateVideoPreview` component to render video blocks in a swiper carousel using `video.js` library ([link](https://github.com/JesusFilm/core/pull/2002/files?diff=unified&w=0#diff-f74580359ca7fd69a7b437b5a2d0f5faf5a111484e0bda91561fc0c56452ae17R1-R135), [link](https://github.com/JesusFilm/core/pull/2002/files?diff=unified&w=0#diff-ab544d12dd7d0706414563ae5511b7c5a13e4dee8248e06c06bda332e4d6195eR1))
*  Compute `videoBlocks` array by filtering and casting journey blocks ([link](https://github.com/JesusFilm/core/pull/2002/files?diff=unified&w=0#diff-501634969741d25ebfcafc40c5ccc588acb4eaaeb397d9b5d56f541f7a38c08fR38-R43))
*  Add a new tab and panel to display `TemplateVideoPreview` with `videoBlocks` prop ([link](https://github.com/JesusFilm/core/pull/2002/files?diff=unified&w=0#diff-501634969741d25ebfcafc40c5ccc588acb4eaaeb397d9b5d56f541f7a38c08fL51-R79))
*  Remove `pointerEvents` style from existing tab to enable interaction ([link](https://github.com/JesusFilm/core/pull/2002/files?diff=unified&w=0#diff-501634969741d25ebfcafc40c5ccc588acb4eaaeb397d9b5d56f541f7a38c08fL51-R79))

### CODE REVIEW

**Review by tataihono — CHANGES_REQUESTED** [review]

File: libs/locales/en/apps-journeys-admin.json [hunk 1]
```diff
@@ -311,7 +311,10 @@
   "Try using fewer filters or look below for templates related to the categories you've selected to search": "Try using fewer filters or look below for templates related to the categories you've selected to search",
   "Use This Template": "Use This Template",
   "Use this template to make it your journey": "Use this template to make it your journey",
-  "{{cardBlockCount}} Cards": "{{cardBlockCount}} Cards",
+  "{{count}} Cards_one": "{{count}} Cards",
```
Comment (tataihono) [inline libs/locales/en/apps-journeys-admin.json:314]:
```suggestion
  "{{count}} Cards_one": "{{count}} Card",
```

File: libs/locales/en/apps-journeys-admin.json [hunk 2]
```diff
@@ -311,7 +311,10 @@
...
+  "{{count}} Cards_other": "{{count}} Cards",
+  "{{count}} Videos_one": "{{count}} Videos",
```
Comment (tataihono) [inline libs/locales/en/apps-journeys-admin.json:316]:
```suggestion
  "{{count}} Videos_one": "{{count}} Video",
```

File: apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx [hunk 3]
```diff
@@ -44,17 +66,27 @@ export function TemplatePreviewTabs(): ReactElement {
...
         <Tab
-          disableRipple
-          label={t('{{cardBlockCount}} Cards', {
-            cardBlockCount: steps?.length ?? 0
+          label={t('{{count}} Cards', {
+            count: steps?.length ?? 0
           })}
           {...tabA11yProps('cards-preview-tab', 0)}
-          sx={{ width: smUp ? 200 : undefined, pointerEvents: 'none' }}
+          sx={{ width: smUp ? 200 : undefined }}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx:73]: no smUp please

File: apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx [hunk 4]
```diff
@@ -44,17 +66,27 @@ export function TemplatePreviewTabs(): ReactElement {
...
+        />
+        <Tab
+          disabled={videos?.length === 0 || videos == null}
+          label={t('{{count}} Videos', {
+            count: videos?.length ?? 0
+          })}
+          {...tabA11yProps('videos-preview-tab', 1)}
+          sx={{ width: smUp ? 200 : undefined }}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx:81]: no smUp please.

File: apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.spec.tsx
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.spec.tsx:1]: remove react-i18next mock

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx [hunk 5]
```diff
@@ -44,17 +64,27 @@ export function TemplatePreviewTabs(): ReactElement {
...
-          sx={{ width: smUp ? 200 : undefined, pointerEvents: 'none' }}
+          sx={{ width: { xs: undefined, sm: 200 } }}
+        />
+        <Tab
+          disabled={videos?.length === 0 || videos == null}
+          label={t('{{count}} Videos', {
+            count: videos?.length ?? 0
+          })}
+          {...tabA11yProps('videos-preview-tab', 1)}
+          sx={{ width: { xs: undefined, sm: 200 } }}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx:79]:
```suggestion
          sx={{ width: { sm: 200 } }}
```

File: apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx [hunk 6]
```diff
@@ -44,17 +64,27 @@ export function TemplatePreviewTabs(): ReactElement {
...
         <Tab
-          disableRipple
-          label={t('{{cardBlockCount}} Cards', {
-            cardBlockCount: steps?.length ?? 0
+          label={t('{{count}} Cards', {
+            count: steps?.length ?? 0
           })}
           {...tabA11yProps('cards-preview-tab', 0)}
-          sx={{ width: smUp ? 200 : undefined, pointerEvents: 'none' }}
+          sx={{ width: { xs: undefined, sm: 200 } }}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx:71]:
```suggestion
          sx={{ width: { sm: 200 } }}
```

Why does this need to be set at all?

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx (hunk 6 above)
Comment (Kneesal) [reply apps/journeys-admin/src/components/TemplateView/TemplatePreviewTabs/TemplatePreviewTabs.tsx:71]: to match design

**Review by tataihono — APPROVED** [review]

**Pull request merged**